src/challenges.js

```diff
@@ -86,17 +197,36 @@ const limitFunctionCallCount = (cb, n) => {
 // Challenge 2: Go back to your class Cube and add the following property: isCube.
 // Create a method inside of Cube that checks for isCube and if it's true, returns a string 'We have a cube!';
 
+Cube.prototype.checkIfCube = function checkIfCube() {
+  if (this.isCube || this instanceof Cube) return 'We have a cube!';
+};
+
 // Use these logs to test your results:
-// console.log(cuboid.volume()); // 100
-// console.log(cuboid.surfaceArea()); // 130
-// console.log(cube.volume()); // 8
-// console.log(cube.surfaceArea()); // 24
-// console.log(cube.checkIfCube());  // "We have a cube!"
+console.log(cuboid.volume()); // 100
+console.log(cuboid.surfaceArea()); // 130
+console.log(cube.volume()); // 8
+console.log(cube.surfaceArea()); // 24
+console.log(cube.checkIfCube());  // "We have a cube!"
 
 // Challenge 3: Recursion
 const checkMatchingLeaves = obj => {
```

Comment:
WAHOOOO! GRATZ ON THIS STRETCH! It is really difficult and I'm so happy you were able to solve it! If at all, your solution is one of the cleaner ones I have seen. Hope that imposter syndrome is close to none because you are doing great!